tests/test_engine_initialization.py

```
@@ -148,6 +148,47 @@ async def _collector(chunk: str, message_type: str = "assistant") -> None:
+):
+    engine = Engine(
+        EngineSettings(),
+        MagicMock(),
+        MagicMock(),
+        MagicMock(),
+        MagicMock(),
+    )
+
+    streamed: list[tuple[str, str]] = []
+
+    async def _fake_get_response(messages, stream=None, stream_callback=None, **kwargs):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_fake_get_response`

Add return type annotation: `str`

(ANN202)